File: ConnectionManagement.cs
using System;
using System.Data;
using System.Data.SQLite;

namespace FamAlbum
{

    public class ConnectionManager
    {
        private string _connectionString;

        public ConnectionManager(string connectionString)
        {
            _connectionString = connectionString;
        }
        public string GetConnString()
        {
            return _connectionString;
        }
        //public SQLiteConnection GetConnection()
        //{
        //    var connection = new SQLiteConnection(_connectionString);
        //    connection.Open();
        //    return connection;
        //}

        public SQLiteConnection GetConnection()
        {
            int attempts = 0;
            while (attempts < 3)
            {
                try
                {
                    var connection = new SQLiteConnection(_connectionString);
                    connection.Open();
                    return connection;
                }
                catch (Exception ex)
                {
                    attempts++;
                    System.Threading.Thread.Sleep(200); // brief pause
                    if (attempts == 3)
                        throw new Exception($"Failed to open SQLite connection after 3 attempts: {ex.Message}", ex);
                }
            }
            return null; // unreachable, but required
        }
    }
    }

File: ConnectionManagement.vb
Imports System.Data.SQLite

Public Class ConnectionManager
    Private _connectionString As String

    Public Sub New(connectionString As String)
        _connectionString = connectionString
    End Sub
    Public Function GetConnString() As String
        Return _connectionString
    End Function
    Public Function GetConnection() As SQLiteConnection
        Dim connection As New SQLiteConnection(_connectionString)
        If connection.State <> ConnectionState.Open Then
            connection.Open()
        End If
        Return connection
    End Function
End Class



